frontend/src/pages/projects/screens/detail/storage/ManageStorageModal.tsx

```diff
@@ -73,60 +79,59 @@ const ManageStorageModal: React.FC<AddStorageModalProps> = ({ existingData, isOp
 
     const pvc = assemblePvc(name, namespace, description, size);
 
-    const handleError = (e: Error) => {
-      setError(e);
-      setActionInProgress(false);
-    };
-    const handleNotebookNameConnection = (pvcName: string) => {
-      if (notebookName) {
-        attachNotebookPVC(notebookName, namespace, pvcName, mountPath.value)
+    const runPromiseActions = async (dryRun: boolean) => {
+      const pvcPromises: Promise<PersistentVolumeClaimKind | NotebookKind>[] = [];
+      if (existingData) {
+        const pvcName = existingData.metadata.name;
+        if (getPvcDisplayName(existingData) !== createData.nameDesc.name) {
+          pvcPromises.push(
+            updatePvcDisplayName(pvcName, namespace, createData.nameDesc.name, { dryRun }),
+          );
+        }
+        if (getPvcDescription(existingData) !== createData.nameDesc.description) {
+          pvcPromises.push(
+            updatePvcDescription(pvcName, namespace, createData.nameDesc.description, { dryRun }),
+          );
+        }
+        if (getPvcTotalSize(existingData) !== createData.size) {
```

Comment:
If a previous request was made but the status hasn't yet updated, we are still making a patch request for the exact same request.